### PR TITLE
[IMP] l10n_in_edi*: quick fix

### PR DESCRIPTION
**Pull request by bat-odoo**
l10n_in_edi:
before updating the JSON value we check the partner country code
for overseas customers.

l10n_in_edi_ewaybill:
Currently, when we select Transportation Mode with Rail, Air or Ship and try
to generate an e-waybill the system raises an error. because of using the wrong fields.

Also made some improvements:
- updated error message
- updated timeout params for jsonrpc because, in the test server, it takes around 61 seconds so we set timeout to 70 seconds.
- filter E-waybill Document Type based on the invoice type

### CODE REVIEW

**Review by jva-odoo — APPROVED** [review]
LGTM

File: addons/l10n_in_edi_ewaybill/models/account_edi_format.py [hunk 1]
```diff
@@ -520,10 +520,10 @@ def _l10n_in_edi_irn_ewaybill_get(self, company, irn):
     @api.model
     def _l10n_in_edi_ewaybill_no_config_response(self):
         return {"error": [{
-            "code": "000",
+            "code": "0",
             "message": _(
-                "A username and password still needs to be set or it's wrong for the E-waybill(IN). "
-                "It needs to be added and verify in the Settings."
+                "Unable to send E-waybill."
```
Comment (h4818) [inline addons/l10n_in_edi_ewaybill/models/account_edi_format.py:525]: .po files

File: addons/l10n_in_edi_ewaybill/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -13,7 +13,7 @@
                             <t class="o_form_label">Setup E-Waybill</t>
                             <span class="fa fa-lg fa-building-o" title="Values set here are company-specific." aria-label="Values set here are company-specific." groups="base.group_multi_company" role="img"/>
                             <div class="text-muted">
-                                Setup E-Waybill Service for this company
+                                Check the <a href="https://www.odoo.com/documentation/15.0/applications/finance/accounting/fiscal_localizations/localizations/india.html">documentation</a> to get credentials
```
Comment (h4818) [inline addons/l10n_in_edi_ewaybill/views/res_config_settings_views.xml:16]: should the link not be version specific

File: addons/l10n_in_edi_ewaybill/models/ewaybill_type.py [hunk 3]
```diff
@@ -30,3 +31,11 @@ def name_get(self):
             name += _(" (Sub-Type: %s)", ewaybill_type.sub_type)
             result_dict[ewaybill_type.id] = name
         return list(result_dict.items())
+
+    @api.model
+    def _search(self, domain, offset=0, limit=None, order=None, count=False, access_rights_uid=None):
+        context = self._context or {}
+        search_mode = context.get('l10n_in_type_search_mode')
+        if isinstance(search_mode, str) and search_mode in ('in', 'out'):
```
Comment (h4818) [inline addons/l10n_in_edi_ewaybill/models/ewaybill_type.py:39]: does it need the `isinstance`?
`if search_mode in ('in', out')` seems ok to me

File: addons/l10n_in_edi_ewaybill/views/account_move_views.xml [hunk 4]
```diff
@@ -16,14 +16,7 @@
...
                             attrs="{'invisible': [('l10n_in_edi_ewaybill_direct_api', '!=', True)]}">
-                            <field name="l10n_in_type_id"
-                                attrs="{'invisible': [
-                                    ('move_type','in',('in_invoice','in_refund','in_receipt'))]}"
-                                domain="[('allowed_supply_type','in',('in','both'))]"/>
-                            <field name="l10n_in_type_id"
-                                attrs="{'invisible': [
-                                    ('move_type','in',('out_invoice','out_refund','out_receipt'))]}"
-                                domain="[('allowed_supply_type','in',('out','both'))]"/>
+                            <field name="l10n_in_type_id" context="{'l10n_in_type_search_mode': (context.get('default_move_type', 'entry') in ('out_invoice', 'out_refund', 'out_receipt') and 'out') or (context.get('default_move_type', 'entry') in ('in_invoice', 'in_refund', 'in_receipt') and 'in') or False}"/>
```
Comment (h4818) [inline addons/l10n_in_edi_ewaybill/views/account_move_views.xml:19]: will this work without a module update?

**Review by h4818 — COMMENTED** [review]
@jco-odoo

File: addons/l10n_in_edi_ewaybill/models/ewaybill_type.py (hunk 3 above)
Comment (jco-odoo) [reply addons/l10n_in_edi_ewaybill/models/ewaybill_type.py:39]: Not sure what the l10n_in_type_search_move is good for.

File: addons/l10n_in_edi_ewaybill/views/res_config_settings_views.xml (hunk 2 above)
Comment (bat-odoo) [reply addons/l10n_in_edi_ewaybill/views/res_config_settings_views.xml:16]: @h4818  As I see documentation argument is used in the master.
I don't know, it's available in V15 and V16?

File: addons/l10n_in_edi_ewaybill/views/account_move_views.xml (hunk 4 above)
Comment (jva-odoo) [reply addons/l10n_in_edi_ewaybill/views/account_move_views.xml:19]: no, and it's just merged so no worry about that

**Comment by jco-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**